orbstation/antagonists/traitor/objectives/final_objective/malware_injection.dm

```diff
@@ -0,0 +1,417 @@
+		return
+	if(injecting)
+		to_chat(user, span_alert("ERROR: Malware injection in progress."))
+		return
+	if(!injecting && corruption_stage && !resuming) // the injection had been aborted at some point
+		user.visible_message(span_danger("[user] hunches over [src] and begins typing away!"), span_notice("You start typing on [src] to resume the malware injection."))
+		resuming = TRUE
+		if(!do_after(user, INJECTOR_HACK_TIME, src, IGNORE_SLOWDOWNS, extra_checks = CALLBACK(src, .proc/inject_check)))
+			resuming = FALSE
+			return
+		resuming = FALSE
+		to_chat(user, span_boldnotice("Resuming malware injection."))
+		// time to go loud!
+		trigger_alarm(user) // set off the alarm, close and lock all doors in the area, alert the AI in big bold letters
+		announce_corruption()
+		advance_corruption_stage(user)
+		return
+	current = select_active_ai(user) // z-level does not matter for this
+	if(!current)
+		to_chat(user, span_alert("No active AIs detected!"))
+	else
+		to_chat(user, span_notice("[current.name] selected for malware injection."))
+
+/obj/machinery/computer/malware/attackby(obj/item/O, mob/living/user, params)
+	if(istype(O, /obj/item/malware_injector))
+		var/obj/item/malware_injector/M = O
+		if(machine_stat & (NOPOWER|BROKEN|MAINT))
+			return
+		if(!istype(user))
+			return
+		if(!user.mind.has_antag_datum(/datum/antagonist/traitor))
+			to_chat(user, span_warning("You have no idea how to use [M]."))
+			return
+		if(M.spent)
+			to_chat(user, span_alert("[M] is no longer functional."))
+			return
+		if(M.hacking)
+			to_chat(user, span_alert("[M] is already hacking something else."))
+			return
+		if(length(M.hacked_apcs) < APCS_REQUIRED_FOR_MALF)
+			to_chat(user, span_alert("[M] has not hacked enough APCs to inject the malware. [APCS_REQUIRED_FOR_MALF - length(M.hacked_apcs)] more needed."))
+			return
+		if(!current)
+			to_chat(user, span_alert("You haven't selected anything to inject malware into!"))
+			return
+		if(!inject_check())
+			to_chat(user, span_alert("Upload failed! Check to make sure [current.name] is functioning properly."))
+			current = null
+			return
+		var/area/comp_area = get_area(src)
+		var/area/req_area = get_req_area(user)
+		if(!req_area) // you don't have the malware injection objective
+			to_chat(user, span_alert("Access denied."))
+			return
+		if(!istype(comp_area, req_area))
+			to_chat(user, span_alert("Upload failed! Unable to establish connection to [current.name] in this area."))
+			return
+		begin_injection(M, user)
+	else
+		return ..()
+
+/obj/machinery/computer/malware/screwdriver_act(mob/living/user, obj/item/I)
+	if(injecting)
+		to_chat(user, span_alert("ERROR: Malware injection in progress."))
+		return FALSE
+	return ..()
+
+/obj/machinery/computer/malware/proc/get_req_area(mob/living/user)
+	if(!user.mind)
+		return
+
+	for(var/datum/traitor_objective/final/malware_injection/malware_objective in SStraitor.taken_objectives_by_type[/datum/traitor_objective/final/malware_injection])
```

Comment:
I don't like this very much but I will admit the only other solution I can think of is passing this property around various objects when you instantiate the hack tool, which isn't a lot better

Reply:
The problem is that the computer needs to have a reference to the user in order to check their mind for the objective, and I  don't know if I can do that on the creation of the computer. I could change it to store the required area as a var and only do this for loop once?

Reply:
How about putting the reference in the injector instead?